pkg/model/serializer.go

```diff
@@ -734,22 +791,16 @@ func (d *Dev) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	err := unmarshal(mode)
 	if err != nil {
 		if mode.Mode == constants.OktetoHybridModeFieldValue {
-			type hybridModeInfo struct {
-				Workdir     string            `json:"workdir,omitempty" yaml:"workdir,omitempty"`
-				Selector    Selector          `json:"selector,omitempty" yaml:"selector,omitempty"`
-				Forward     []forward.Forward `json:"forward,omitempty" yaml:"forward,omitempty"`
-				Environment Environment       `json:"environment,omitempty" yaml:"environment,omitempty"`
-				EnvFiles    EnvFiles          `json:"envFiles,omitempty" yaml:"envFiles,omitempty"`
-				Command     Command           `json:"command,omitempty" yaml:"command,omitempty"`
-				Reverse     []Reverse         `json:"reverse,omitempty" yaml:"reverse,omitempty"`
-				Mode        string            `json:"mode,omitempty" yaml:"mode,omitempty"`
-			}
-
 			hybridModeDev := &hybridModeInfo{}
 			err := unmarshal(hybridModeDev)
 			if err != nil {
 				return err
 			}
+
+			warningMsg := hybridModeDev.warnHybridUnsupportedFields()
+			if warningMsg != "" {
+				oktetoLog.Warning(warningMsg)
```

Comment:
i am thinking maybe we could just call the function that shows or not the warning instead of checking the output.

`warnHybridUnsupportedFields` i feel the function is going to warn, if we are returning output and then warning i would change naming maybe `getHybridUnsupportedFields` and with those then warn? instead of returning the message, return the fields and then create the message out. WDYT?

Reply:
@teresaromero we need to keep `oktetoLog.Warning` calls outside of the function so that it's easily testable.

I see your point about the naming, I am open for suggestions! I think `getHybridUnsupportedFields` implies it returns the actual fields? The word "warn" in the current one makes me think about logging/warnings and having the return value called `warningMsg` feels quite clear.

I can't think of alternatives now but if you have one, happy to follow-up on this 🙏